source/flux/parts/settings.js

```diff
@@ -83,20 +84,42 @@ export function logInViaCredentials(
   username: string,
   password: string,
 ): ThunkAction<LogInActions | UpdateBalancesType> {
-  return async dispatch => {
+  return async (dispatch, getState) => {
     dispatch({type: CREDENTIALS_LOGIN_START})
-
+    const state = getState()
+    const isConnected = state.app ? state.app.isConnected : false
+    if (!isConnected) {
+      dispatch({type: CREDENTIALS_LOGIN_FAILURE})
+      showNetworkFailureMessage()
+    }
     const result = await performLogin(username, password)
     if (result) {
       dispatch({type: CREDENTIALS_LOGIN_SUCCESS, payload: {username, password}})
       // since we logged in successfully, go ahead and fetch the meal info
       dispatch(updateBalances())
     } else {
       dispatch({type: CREDENTIALS_LOGIN_FAILURE})
+      showInvalidLoginMessage()
     }
   }
 }
 
+function showNetworkFailureMessage() {
```

Comment:
I don't mind having these in this file; however, given that we only call them once, what about inlining them into the logInViCredentials function itself?

Reply:
Is this where they should be? Or should they be above the return function? https://github.com/StoDevX/AAO-React-Native/pull/2019/commits/d1f2b9d71a733dc4d391d66d51dbbee20e4e6e4f